tesseract-server/src/handlers/logic_layer/aggregate.rs

```diff
@@ -71,124 +133,188 @@ pub fn logic_layer_aggregation(
         static ref QS_NON_STRICT: qs::Config = qs::Config::new(5, false);
     }
 
-    let mut cube_name;
-
-    let mut agg_query = match QS_NON_STRICT.deserialize_str::<LogicLayerQueryOpt>(query) {
-        Ok(mut q) => {
-            // Check to see if the logic layer config has a alias with the
-            // provided cube name
-            cube_name = match logic_layer_config.clone() {
-                Some(llc) => {
-                    match llc.sub_cube_name(q.cube.clone()) {
-                        Ok(cn) => cn,
-                        Err(_) => q.cube.clone()
-                    }
-                },
-                None => q.cube.clone()
-            };
+    let agg_query = match QS_NON_STRICT.deserialize_str::<LogicLayerQueryOpt>(query) {
+        Ok(q) => q,
+        Err(err) => return boxed_error(err.to_string())
+    };
 
-            let cube = match schema.get_cube_by_name(&cube_name) {
-                Ok(c) => c.clone(),
-                Err(err) => return boxed_error(err.to_string())
-            };
+    // Check to see if the logic layer config has a alias with the
+    // provided cube name
+    let cube_name = match logic_layer_config.clone() {
+        Some(llc) => {
+            match llc.sub_cube_name(agg_query.cube.clone()) {
+                Ok(cn) => cn,
+                Err(_) => agg_query.cube.clone()
+            }
+        },
+        None => agg_query.cube.clone()
+    };
 
-            let cube_cache = req.state().cache.read().unwrap().find_cube_info(&cube_name).clone();
+    let cube = match schema.get_cube_by_name(&cube_name) {
```

Comment:
I forget if `get_cube_by_name` returns a reference to a cube. But if it did, would it be useable for your purposes in the rest of this method?

Reply:
It does return a reference. Are you just saying I should remove the clone?

Reply:
is it possible to remove the clone? I guess if you're just cloning a ref it's not as big a deal. But it did give me pause.

Reply:
I already removed the clone here: https://github.com/tesseract-olap/tesseract/pull/162/commits/c9751375aca7e20d494c6e3a877c136d601d82fb#diff-a1b1aab7cc043def804944a357fb1fedL154